### PR TITLE
Settings in vue

### PR DESCRIPTION
**Pull request by jotoeri**
Argh, sorry, @rullzer,
as i was just motivated, i did another vue-migration this night and just realised afterwards you gave it a go already. 🤦‍♂️ So this PR is just to show what i have, i leave it up to you to decide if you want to continue on your approach or if i should. It anyways was a nice experience for me to do this. 👍 

So this version is in fact working version already. Probably to fix:
- I used `OCP.AppConfig.setValue` and `OC.msg....` now, as it was before. Seems like your ansatz was an API and axios?
- From the other comment, seems like i'd need to commit the js-files?

### CODE REVIEW

File: src/AdminSettings.vue [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div id="password-policy" class="section">
```
Comment (skjnldsv) [inline src/AdminSettings.vue:25]: https://nextcloud-vue-components.netlify.app/#/Components/SettingsSection

File: src/AdminSettings.vue [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+					message = t(appName, 'Minimal length has to be a non negative number')
+					break
+				case 'historySize':
+					message = t(appName, 'History size has to be a non negative number')
+					break
+				case 'expiration':
+					message = t(appName, 'Expiration days have to be a non negative number')
+					break
+				case 'maximumLoginAttempts':
+					message = t(appName, 'Maximum login attempts have to be a non negative number')
```
Comment (skjnldsv) [inline src/AdminSettings.vue:156]: I'm not sure that would be caught by the translate bot.
I think it needs to be set with the hardcoded app name

Comment (jotoeri) [reply src/AdminSettings.vue:156]: Ahh, sure, you're right. 🤦‍♂️

File: src/AdminSettings.vue (hunk 1 above)
Comment (jotoeri) [reply src/AdminSettings.vue]: Changed that. 👍 
However, it increases the size of the bundle for that small section from `100KiB` to `330KiB` 🤔 

The SettingsSection is missing the small border at the bottom. Is that a planned new design or a but on components? 🤔

**Comment by jotoeri** [comment]
Btw. also fixes #114

File: src/AdminSettings.vue (hunk 1 above)
Comment (skjnldsv) [reply src/AdminSettings.vue]: > * However, it increases the size of the bundle for that small section from `100KiB` to `330KiB` thinking
> *

We should definitely investigate what weights that much then

Comment (jotoeri) [reply src/AdminSettings.vue]: Hm, seems to be:
- 140K of l10n in component (with another bug, that the single string that wants to use it, is not extracted)
- 30K Iconfonts
- 30K of an additional separate gettext lib

grafik
grafik

**Comment by rullzer** [comment]
Awesome stuff! Great that you did it. Much more complete than mine it seems. So. Let's continue here.

File: src/AdminSettings.vue (hunk 1 above)
Comment (jotoeri) [reply src/AdminSettings.vue]: So:
- l10n is tackled on vue-components
- inconfonts need to be separated somehow on vue-components
- gettext/plurals also seems to come from vue-components: if i bundle with a linked local repo, then it is listed within the @nextcloud/vue section.

@skjnldsv Can we resolve this issue here? It's not ideal, but should at least be down to 230 KB bundle-size for now.

File: webpack.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const { merge } = require('webpack-merge')
+const path = require('path')
+const webpackConfig = require('@nextcloud/webpack-vue-config')
+
+const config = {
+	entry: {
+		settings: path.resolve(path.join('src', 'settings.js'))
```
Comment (skjnldsv) [inline webpack.js:7]: Will most likely throw an error stating the 'main' entry file doesn't exists?

Comment (jotoeri) [reply webpack.js:7]: Just have a look 5 lines below... 😉 
```
delete mergedConfig.entry.main
```

Comment (jotoeri) [reply webpack.js:7]: Next step will then be to have a second entry `settings-sharing`, which is why i chose this way instead of using the main-entry.

**Review by skjnldsv — APPROVED** [review]

**Comment by jotoeri** [comment]
@rullzer, @ChristophWurst somebody up for a second review here? 😊

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,7 @@
 /vendor/
 .php_cs.cache
+
+node_modules/
+
+# build dir
+js/
```
Comment (rullzer) [inline .gitignore:7]: Assets ahve to be commited here as we use it during packaging

**Pull request merged**